introspective_ORB_SLAM/include/io_access.h

```diff
@@ -184,66 +203,71 @@ class PFM {
-      * Since that's the same format that most CPUs and compiler use,
-      * you can usually just make a program use the bytes directly
-      * as a floating point number, after taking care of the
-      * endianness variation.
-      */
-    int length_ = this->width*this->height;
-    this->endianess = endianess_;
-    T* tbimg = (T *)malloc(length_ * sizeof(T));
+     * The raster is a sequence of pixels, packed one after another,
+     * with no delimiters of any kind. They are grouped by row,
+     * with the pixels in each row ordered left to right and
+     * the rows ordered bottom to top.
+     * Each pixel consists of 1 or 3 samples, packed one after another,
+     * with no delimiters of any kind. 1 sample for a grayscale PFM
+     * and 3 for a color PFM (see the Identifier Line of the PFM header).
+     * Each sample consists of 4 consecutive bytes. The bytes represent
+     * a 32 bit string, in either big endian or little endian format,
+     * as determined by the Scale Factor / Endianness line of the PFM
+     * header. That string is an IEEE 32 bit floating point number code.
+     * Since that's the same format that most CPUs and compiler use,
+     * you can usually just make a program use the bytes directly
+     * as a floating point number, after taking care of the
+     * endianness variation.
+    */
+
+    int length = this->width_ * this->height_;
+    this->endianess_ = endianess;
+    T* tbimg = (T *)malloc(length * sizeof(T));
     // PFM SPEC image stored bottom -> top reversing image
-    for (int i = 0; i < this->height; i++) {
-        memcpy(&tbimg[(this->height - i - 1)*this->width],
-            &imgbuffer[(i*this->width)],
-            this->width * sizeof(T));
+    for( int i = 0; i < this->height_; ++i ) 
+    {
+      memcpy( &tbimg[ (this->height_ - i - 1)*this->width_ ],
+              &imgbuffer[ (i * this->width_) ],
+              this->width_ * sizeof(T) );
     }
 
-    if (this->is_little_big_endianness_swap()) {
-        std::cout << "little-big endianness transformation is needed.\n";
-        // little-big endianness transformation is needed.
-        union {
-            T f;
-            unsigned char u8[sizeof(T)];
-        } source, dest;
+    if( this->is_little_big_endianness_swap() ) 
+    {
+      std::cout << "little-big endianness transformation is needed.\n";
+      // little-big endianness transformation is needed.
+      union{ T f; unsigned char u8[sizeof(T)]; } source, dest;
 
-        for (int i = 0; i < length_; ++i) {
-            source.f = tbimg[i];
-            for (size_t k = 0, s_T = sizeof(T); k < s_T; k++)
-                dest.u8[k] = source.u8[s_T - k - 1];
-            tbimg[i] = dest.f;
-            //cout << dest.f << ", ";
+      for (int i = 0; i < length; ++i) 
+      {
+        source.f = tbimg[i];
+
+        for (size_t k = 0, s_T = sizeof(T); k < s_T; k++)
+        {
+          dest.u8[k] = source.u8[s_T - k - 1];
         }
```

Comment:
I think that even single line `for` statements should be in scope brackets. I know it adds an extra two lines but it will prevent simple mistakes down the line (made by me lol)